packages/framework-integration-tests/integration/end-to-end/cart.integration.ts

```diff
@@ -325,6 +325,175 @@ describe('Cart end-to-end tests', () => {
         expect(productData).to.be.deep.equal(expectedResult)
       })
     })
+
+    context('Deletions', () => {
+      let mockSku: string
+      let mockDisplayName: string
+      let mockDescription: string
+      let mockPriceInCents: number
+      let mockCurrency: string
+
+      let productId: string
+
+      beforeEach(async () => {
+        mockSku = random.uuid()
+        mockDisplayName = commerce.productName()
+        mockDescription = lorem.paragraph()
+        mockPriceInCents = random.number({ min: 1 })
+        mockCurrency = finance.currencyCode()
+
+        await client.mutate({
+          variables: {
+            sku: mockSku,
+            displayName: mockDisplayName,
+            description: mockDescription,
+            priceInCents: mockPriceInCents,
+            currency: mockCurrency,
+          },
+          mutation: gql`
+            mutation CreateProduct(
+              $sku: String!
+              $displayName: String
+              $description: String
+              $priceInCents: Float
+              $currency: String
+            ) {
+              CreateProduct(
+                input: {
+                  sku: $sku
+                  displayName: $displayName
+                  description: $description
+                  priceInCents: $priceInCents
+                  currency: $currency
+                }
+              )
+            }
+          `,
+        })
+
+        // Check that latest update product is available in read model
+        const products = await waitForIt(
+          () => {
+            return client.query({
+              query: gql`
+                query {
+                  ProductReadModels {
+                    id
+                    sku
+                    displayName
+                    description
+                    price
+                    availability
+                    deleted
+                  }
+                }
+              `,
+            })
+          },
+          (result) => result?.data?.ProductReadModels?.some((product: any) => product.sku === mockSku)
+        )
+
+        const product = products.data.ProductReadModels.find((product: any) => product.sku === mockSku)
+        productId = product.id
+      })
+
+      it('should delete the entity as expected', async () => {
+        const MAX_UPDATES = 100
+        const adminEmail: string = internet.email()
```

Comment:
I believe that the type annotation is not needed:
```suggestion
        const adminEmail = internet.email()
```